lib/LanguageServerRename/Model/Renamer.php

```diff
@@ -0,0 +1,314 @@
+use Microsoft\PhpParser\Node\Expression\ScopedPropertyAccessExpression;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\Parameter;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Node\QualifiedName;
+use Microsoft\PhpParser\Node\Statement\ClassDeclaration;
+use Microsoft\PhpParser\Node\Statement\InterfaceDeclaration;
+use Microsoft\PhpParser\Node\Statement\TraitDeclaration;
+use Microsoft\PhpParser\Node\UseVariableName;
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\Token;
+use Phpactor\CodeTransform\Domain\Refactor\RenameVariable;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\Extension\LanguageServerBridge\Converter\PositionConverter;
+use Phpactor\LanguageServerProtocol\Position;
+use Phpactor\LanguageServerProtocol\Range;
+use Phpactor\LanguageServerProtocol\RenameFile;
+use Phpactor\LanguageServerProtocol\TextDocumentEdit;
+use Phpactor\LanguageServerProtocol\TextDocumentItem;
+use Phpactor\LanguageServerProtocol\TextEdit;
+use Phpactor\LanguageServerProtocol\VersionedTextDocumentIdentifier;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+use Phpactor\ReferenceFinder\DefinitionLocator;
+use Phpactor\ReferenceFinder\Exception\CouldNotLocateDefinition;
+use Phpactor\ReferenceFinder\ReferenceFinder;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\Location;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use function mb_strlen;
+use function mb_substr;
+
+class Renamer
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+    /**
+     * @var DefinitionLocator
+     */
+    private $definitionLocator;
+    /**
+     * @var ReferenceFinder
+     */
+    private $finder;
+    /**
+     * @var ClientApi
+     */
+    private $clientApi;
+    /**
+     * @var int
+     */
+    private $timeoutSeconds = 10;
+    /**
+     * @var Workspace
+     */
+    private $workspace;
+    /**
+     * @var RenameVariable
+     */
+    private $renameVariable;
+    /**
+     * @var NodeUtils
+     */
+    private $nodeUtils;
+
+    public function __construct(
+        Workspace $workspace,
+        Parser $parser,
+        ReferenceFinder $finder,
+        DefinitionLocator $definitionLocator,
+        ClientApi $clientApi,
+        RenameVariable $renameVariable,
+        NodeUtils $nodeUtils
+    ) {
+        $this->parser = $parser;
+        $this->definitionLocator = $definitionLocator;
+        $this->finder = $finder;
+        $this->clientApi = $clientApi;
+        $this->workspace = $workspace;
+        $this->renameVariable = $renameVariable;
+        $this->nodeUtils = $nodeUtils;
+    }
+
+    public function prepareRename(TextDocumentItem $textDocument, Position $position): ?Range
```

Comment:
public APIs should accept Phpactor documents, in this case the text document and range from the Phpactor `phpactor/text-document` library. This would enable us to use this service outside of the LS (CLI / RPC)

Reply:
Yep, for the next version.